cluster/addons/dns/skydns-rc.yaml.in

```diff
@@ -19,6 +19,9 @@ spec:
         version: v11
         kubernetes.io/cluster-service: "true"
     spec:
+{% if grains['cloud'] is defined and grains['cloud'] in [ 'vsphere', 'photon-controller' ] %}
```

Comment:
Before you roll back, could you look at this pull request and see if it addresses the problem?
https://github.com/kubernetes/kubernetes/pull/24124
It looks like it could be merged, but it hasn't been--I'm not sure why.

The issue this was solving wasn't about knowing which nameservers to contact, it was about actually being able to contact them: without enabling the use of the host network, we were unable to communicate with the DNS server. 

Ultimately, this change was meant to only affect the vSphere and Photon Controller implementations. Clearly I didn't do that correctly. Is there a way we can keep the underlying change without breaking other implementations?